scene/audio/audio_stream_player.cpp

```diff
@@ -106,57 +105,80 @@ void AudioStreamPlayer::set_pitch_scale(float p_pitch_scale) {
 	ERR_FAIL_COND(p_pitch_scale <= 0.0);
 	pitch_scale = p_pitch_scale;
 
-	AudioServer::get_singleton()->set_playback_pitch_scale(stream_playback, pitch_scale);
+	for (Ref<AudioStreamPlayback> &playback : stream_playbacks) {
+		AudioServer::get_singleton()->set_playback_pitch_scale(playback, pitch_scale);
+	}
 }
 
 float AudioStreamPlayer::get_pitch_scale() const {
 	return pitch_scale;
 }
 
+void AudioStreamPlayer::set_max_polyphony(int p_max_polyphony) {
+	if (p_max_polyphony > 0) {
+		max_polyphony = p_max_polyphony;
+	}
+}
+
+int AudioStreamPlayer::get_max_polyphony() const {
+	return max_polyphony;
+}
+
 void AudioStreamPlayer::play(float p_from_pos) {
```

Comment:
I am not sure why this was not checked before, but the node should check whether its inside tree to allow play by doing something like:
```
ERR_FAIL_COND_MSG(!is_inside_tree(),"Playback can only happy when a node is inside the scene tree");
```

Reply:
Added this check.